classes/submission/SubmissionDAO.inc.php

```diff
@@ -471,20 +471,33 @@ function getByUserId($userId, $contextId = null) {
 	 *  whose section will be included in the results (excluding others).
 	 * @param $includeDeclined boolean optional include submissions which have STATUS_DECLINED
 	 * @param $includePublished boolean optional include submissions which are published
+	 * @param $submissionId int|null optional Filter by submission id.
 	 * @param $title string|null optional Filter by title.
 	 * @param $author string|null optional Filter by author.
 	 * @param $stageId int|null optional Filter by stage id.
 	 * @param $sectionId int|null optional Filter by section id.
 	 * @param $rangeInfo DBRangeInfo
 	 * @return DAOResultFactory containing matching Submissions
 	 */
-	function getBySubEditorId($contextId, $subEditorId = null, $includeDeclined = true, $includePublished = true, $title = null, $author = null, $stageId = null, $sectionId = null, $rangeInfo = null) {
+	function getBySubEditorId($contextId, $subEditorId = null, $includeDeclined = true, $includePublished = true, $submissionId = null, $title = null, $author = null, $stageId = null, $sectionId = null, $rangeInfo = null) {
 		$params = $this->getFetchParameters();
 		if ($subEditorId) $params[] = (int) $subEditorId;
 		$params[] = (int) $contextId;
 		$params[] = (int) ROLE_ID_MANAGER;
 		$params[] = (int) ROLE_ID_SUB_EDITOR;
 
+		if ($submissionId) {
```

Comment:
Sorry, yes...

    if ($submissionId) $submissionId = filter_var($submissionId, FILTER_VALIDATE_INT);

...should maybe be...

    if ($submissionId) $params[] = filter_var($submissionId, FILTER_VALIDATE_INT);

This should be OK if an invalid submission ID is entered, because `filter_var` returns `false` in that case, and the result of the `int` cast will be `0`. The check in the SQL side to add the condition will pass (`null !== false`) and the `0` condition will match no submissions, as appropriate.